cookietemple/lint/template_linter.py

```diff
@@ -583,3 +591,93 @@ def init_changelog_lint_content(self) -> list:
+        """
+        Lint the sections from the cookietemple.cfg file applying the following rules:
+
+        1.) Every config file should have at least the following sections:
+            bumpversion, bumpversion_files_whitelisted, bumpversion_files_blacklisted,
+            sync_files_blacklisted, sync_level
+
+        2.) 'bumpversion' should contain a 'current_version' value (the project's current version)
+
+        3.) 'bumpversion_files_whitelisted' should contain at least the '.cookietemple.yml' file
+
+        4.) 'sync_level' should contain a 'ct_sync_level' value (and this value should be one of either 'patch', 'minor' or 'major')
+
+        5.) 'sync_files_blacklisted' should contain at least the 'CHANGELOG.rst' file (excluding it from syncing to avoid PR updates)
+        """
+        self.parser.read(f'{self.config_file_path}')
+        no_section_missing = self.check_missing_sections(self.parser.sections())
+        if no_section_missing:
+            check_section_flag = True
+            check_section_flag &= self.check_section(self.parser.items('bumpversion'), 'bumpversion')
+            check_section_flag &= self.check_section(self.parser.items('bumpversion_files_whitelisted'), 'bumpversion_files_whitelisted')
+            check_section_flag &= self.check_section(self.parser.items('sync_level'), 'sync_level')
+            check_section_flag &= self.check_section(self.parser.items('sync_files_blacklisted'), 'sync_files_blacklisted')
+            if check_section_flag:
+                self.linter_ctx.passed.append(('general-7', 'All config sections passed cookietemple linting!'))
+        else:
+            self.linter_ctx.failed.append(('general-7', 'Aborted config section linting. Fix missing sections first!'))
+
+    def check_missing_sections(self, parsed_sections) -> bool:
+        """
+        Examine cookietemple config file for missing sections
+        :param parsed_sections: All parsed sections from the user's cookietemple.cfg file
+        """
+        sections = ['bumpversion', 'bumpversion_files_whitelisted', 'bumpversion_files_blacklisted', 'sync_files_blacklisted', 'sync_level']
+        missing_sections = []
+        # for every section check whether its in the parsed sections or not
+        for section in sections:
+            if section not in parsed_sections:
+                missing_sections.append(section)
+        # if there were any missing sections, let linter fail
+        if missing_sections:
+            miss_section_info = 'Cookietemple config file misses section' + 's' if len(missing_sections) > 1 else ''
+            self.linter_ctx.failed.append(('general-7', miss_section_info + f': {" ".join(section for section in missing_sections)}'))
+            return False
+        else:
+            self.linter_ctx.passed.append(('general-7', 'All required cookietemple.cfg sections were found!'))
+            return True
+
+    def check_section(self, section_items, section_name) -> bool:
```

Comment:
@Zethson Im not quite happy with those longer if statements and the function design, as their functionality is very similar. You have any ideas on redesigning?

Reply:
Maybe it is possible to use an Ordered inserted dict and then just go through every pair and check whether it is correctly available?
-> `{bumpversion: section0: (current_version, linting-code), section1: (another_version, linting-code)}`

If any of those fail we can just build a message from that

We know which key (= function) failed and we know the exact failure and an error code.
You could e.g. use `general-7-1` and `general-7-2`